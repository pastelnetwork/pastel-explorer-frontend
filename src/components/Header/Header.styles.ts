import styled from 'styled-components';
import { Divider as MuiDivider, Typography as MuiTypography } from '@mui/material';

export const Divider = styled(MuiDivider);

export const Typography = styled(MuiTypography);

export const Container = styled.div`
  margin-bottom: ${props => props.theme.spacing(5)}px;
`;
