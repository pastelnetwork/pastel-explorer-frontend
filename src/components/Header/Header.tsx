import * as React from 'react';
import { Grid } from '@material-ui/core';

import * as Styles from './Header.styles';

interface HeaderProps {
  title: string;
  className?: string;
}

const Header: React.FC<HeaderProps> = ({ title, className = '' }) => {
  return (
    <Styles.Container className={className}>
      <Grid justify="space-between" container>
        <Grid item>
          <Styles.Typography variant="h3" gutterBottom>
            {title}
          </Styles.Typography>
        </Grid>
      </Grid>
      <Styles.Divider my={1} />
    </Styles.Container>
  );
};

export default Header;
