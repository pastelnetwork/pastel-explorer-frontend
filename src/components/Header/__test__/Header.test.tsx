import { shallow } from 'enzyme';

import 'jest-styled-components';

import i18next from '../../../utils/helpers/i18n';
import Header from '../Header';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
});

describe('components/Header', () => {
  const wrapper = shallow(<Header title="Donec a egestas mauris" />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
