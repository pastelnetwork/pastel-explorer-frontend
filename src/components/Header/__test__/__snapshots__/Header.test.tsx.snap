// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header renders correctly 1`] = `ShallowWrapper {}`;
